lib/validator.js

```diff
@@ -4,53 +4,94 @@
 'use strict';
 
 const bedrock = require('bedrock');
-const {callbackify, BedrockError} = bedrock.util;
+const {constants} = require('bedrock').config;
+const logger = require('./logger');
+const serializeError = require('serialize-error');
+const {validate} = require('bedrock-validation');
+const {BedrockError} = bedrock.util;
 
 const api = {};
 module.exports = api;
 
-api.validate = callbackify(async (input, validatorConfigs, {ledgerNode}) => {
+api.validate = async ({ledgerConfiguration, ledgerNode, operation}) => {
+  // if config.get() is null, then the config we are validating
+  // must be treated as the genesis config, so validate against itself
+  const ledgerConfig = (await ledgerNode.config.get()) || ledgerConfiguration;
+  if(!ledgerConfig) {
+    throw new BedrockError(
+      'The ledger has not been configured.', 'InvalidStateError');
+  }
+  let validatorConfigs = [];
+  let validatorInput;
+  if(operation) {
+    // an operation is being validated
+    validatorConfigs = ledgerConfig.operationValidator || [];
+    // validate the basic shape of the operation
+    _validateOperation(operation);
+    validatorInput = operation;
+  }
+  if(ledgerConfiguration) {
+    // a ledgerConfiguration is being validated
+    validatorConfigs = ledgerConfig.ledgerConfigurationValidator || [];
+    validatorInput = ledgerConfiguration;
+  }
   if(validatorConfigs.length === 0) {
     // nothing to validate
     return;
   }
 
+  if(typeof validatorInput !== 'object') {
+    throw new TypeError('validatorInput must be an object.');
+  }
+
   // run all the validators in parallel
-  const validatorReports = [];
+  const promises = [];
   for(const validatorConfig of validatorConfigs) {
-    const report = await _runValidator(input, validatorConfig, {ledgerNode});
-    validatorReports.push(report);
+    promises.push(_runValidator(
+      {ledgerConfig, ledgerNode, validatorConfig, validatorInput}));
   }
+  const validatorReports = await Promise.all(promises);
```

Comment:
Could do: `await Promise.all(validatorConfigs.map(` instead... but just a matter of style really.

Reply:
done.